app/Models/Device.php

```diff
@@ -252,4 +256,46 @@ public function setIpAttribute($ip)
         $this->attributes['ip'] = inet_pton($ip);
     }
 
+    // ---- Query scopes ----
+
+    public function scopeDeviceUp($query)
```

Comment:
I did it like that as we also have the same for ports. IsUp is certainly nicer though, I'll switch over.